graph/step.go

```diff
@@ -196,3 +202,42 @@ func (s *Step) UpdateBuildStepWithDefaults() {
 		s.Build = fmt.Sprintf("--isolation hyperv %s", s.Build)
 	}
 }
+
+// UseBuildCacheForBuildStep indicates if buildx needs to be used
+func (s *Step) UseBuildCacheForBuildStep() bool {
+	if s.EnableCache && s.CacheID == "" {
+		s.CacheID = "cache"
+	}
+
+	return s.IsBuildStep() && s.CacheID != ""
+}
+
+// FetchBuildWithBuildCache returns an updated build step by adding buildx cache parameters
+func (s *Step) FetchBuildWithBuildCache() (string, error) {
+	result := ""
+
+	if s.Tags != nil && len(s.Tags) == 0 {
+		return result, pkgerrors.New("(buildcache) did not provide any tags")
+	}
+
+	buildImg := s.Tags[0]
+	repo, err := reference.Parse(buildImg)
+	if err != nil {
+		return result, pkgerrors.Wrap(err, "(buildcache) failed to parse the buildimage")
+	}
+
+	// buildImg: "b.gcr.io/test.example.com/my-app:test.example.com",
+	// s.CacheID: "myBuildCache"
+	// cacheImage: "b.gcr.io/test.example.com/my-app:myBuildCache"
+	if named, ok := repo.(reference.Named); ok {
+		cacheImage, err := reference.WithTag(named, s.CacheID)
+		if err != nil {
+			return result, pkgerrors.Wrap(err, "(buildcache) failed to attach cacheID tag to the repo")
```

Comment:
nit: the (buildcache) prefix is really uncommon and will look weird to users

Reply:
I ll remove this once i have tested. I have added additional logging for testing purposes